src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,151 @@
+
+        } else if (echo.startsWith("event")) {
+            String eventArguments = echo.substring(5).trim();
+            if (eventArguments.isEmpty()) {
+                throw(new NoTaskDescriptionException("     ☹ OOPS!!! The description of a event cannot be empty."));
+            }
+            String[] splitArguments = eventArguments.split("/");
+            return new Event(splitArguments[0], splitArguments[1].substring(4).trim(),
+                    splitArguments[2].substring(2).trim());
+
+        } else {
+            // System.out.println("Placeholder");
+            throw(new InvalidInputException("      ☹ OOPS!!! I'm sorry, but I don't know what that means :-("));
+        }
+    }
+
+    /**
+     * Returns a Task object corresponding to input String.
+     * Made to be used to read from saved files, hence containing addition argument to check if tasks are done.
+     *
+     * @param echo Input string to be parsed
+     * @param isDone Boolean to check if corresponding created Task is marked as done
+     * @return Task corresponding to given String
+     */
+    public static Task parseFileReader(String echo, boolean isDone) {
+        if (echo.startsWith("todo")) {
+            return new Todo(isDone, echo.substring(4).trim());
+        } else if (echo.startsWith("deadline")) {
+            String deadlineArguments = echo.substring(8).trim();
+            String[] splitArguments = deadlineArguments.split("/");
+            return new Deadline(isDone, splitArguments[0], splitArguments[1].substring(2).trim());
+        } else if (echo.startsWith("event")) {
+            String eventArguments = echo.substring(5).trim();
+            String[] splitArguments = eventArguments.split("/");
+            return new Event(isDone, splitArguments[0], splitArguments[1].substring(4).trim(),
+                    splitArguments[2].substring(2).trim());
+        } else {
+            throw(new InvalidInputException("      ☹ OOPS!!! File format seems weird"));
+        }
+    }
+
+    /**
+     * Returns boolean based on whether a corresponding input string has a non-task creation command.
+     * Attempts to carry out command if command is recognised, and returns true.
+     * Else returns false.
+     *
+     * @param echo Input String to be parsed for commands.
+     * @param tasks TaskList Object which encapsulates the current list of tasks.
+     * @param ui Ui Object to call Ui-related methods.
+     * @return Boolean if any tasks are recognised and carried out.
+     */
+    public static boolean parseCommands(String echo, TaskList tasks, Ui ui) {
+        if (echo.equals("list")) {
+            System.out.println("    OK, Here are the items in your list: ");
+            Ui.printArrayList(tasks.getList());
+            // put in loop to read the list
+            return true;
+        }
+
+        if (echo.startsWith("mark")) {
+            try {
+                int taskToModify = Integer.parseInt(echo.replaceAll("[^0-9]", ""));
```

Comment:
Seems like invalid commands like `mark 1 thetask 0` would still work with this code, perhaps using `.split` or regex could prevent invalid commands from working.